plugins/optimization-detective/storage/rest-api.php

```diff
@@ -169,28 +172,20 @@ function od_handle_rest_request( WP_REST_Request $request ) {
 	$post_id = $result;
 
 	/**
-	 * Fires whenever a URL Metric was successfully collected.
+	 * Fires whenever a URL Metric was successfully stored.
 	 *
-	 * @since 0.6.0
+	 * @since n.e.x.t
 	 *
-	 * @param array $context {
-	 *     Context about the successful URL Metric collection.
-	 *
-	 *     @var int                                   $post_id
-	 *     @var WP_REST_Request<array<string, mixed>> $request
-	 *     @var OD_Strict_URL_Metric                  $url_metric
-	 *     @var OD_URL_Metrics_Group                  $group
-	 *     @var OD_URL_Metrics_Group_Collection       $group_collection
-	 * }
+	 * @param OD_URL_Metric_Stored_Context $context Context.
 	 */
 	do_action(
-		'od_url_metric_collected',
-		array(
-			'post_id'                      => $post_id,
-			'request'                      => $request,
-			'url_metric'                   => $url_metric,
-			'url_metrics_group'            => $group,
-			'url_metrics_group_collection' => $group_collection,
+		'od_url_metric_stored',
+		new OD_URL_Metric_Stored_Context(
+			$request,
+			$post_id,
+			$url_metric_group_collection,
+			$url_metric_group,
+			$url_metric
 		)
```

Comment:
Should the old filter still be applied as deprecated rather than removed?

More importantly though, why wrap the data in a class? Not saying we shouldn't do it, but I'm not sure what's the value, even more so why this needs to happen _now_ (as part of this PR).

Reply:
The old action was just introduced in the most recent version and no plugins are using it yet. So I don't think there is any need for deprecation here. This is in contrast with the other deprecation logic in this PR, as Image Prioritizer and Embed Optimizer are using the changed API surface.

Wrapping the data in a class gives more flexibility to change the property names later if need be, using a similar deprecation strategy with the magic getter in this PR. It also allows for autocompletion for the properties. 

I figured to include it here because I was already changing the `url_metrics_group` and `url_metrics_group_collection` array keys, so it seemed like a logical opportunity to change the names but also switch from an array to an object.